source/adaptive_load/metrics_evaluator_impl.cc

```diff
@@ -15,16 +15,42 @@ using ::nighthawk::adaptive_load::MetricSpec;
 using ::nighthawk::adaptive_load::MetricSpecWithThreshold;
 using ::nighthawk::adaptive_load::ThresholdSpec;
 
+// Extract estimated measuring period from Nighthawk output proto.
+absl::StatusOr<MeasuringPeriod> GetMeasuringPeriod(const nighthawk::client::Output& output) {
+  if (output.results().empty()) {
+    return absl::Status{absl::StatusCode::kInvalidArgument, "output.results cannot be empty."};
+  }
+  // Assume the first result is a good representation for the entire run.
+  MeasuringPeriod time_interval;
+  time_interval.start_time = output.results(0).execution_start();
+  time_interval.duration = output.results(0).execution_duration();
+  return time_interval;
+}
+
+// Utility function for GetMetric functionality with fallback logic.
+absl::StatusOr<double> GetMetric(MetricsPlugin& metrics_plugin, absl::string_view metric_name,
+                                 const MeasuringPeriod& measuring_period) {
+  absl::StatusOr<double> metric_value_or =
+      metrics_plugin.GetMetricByNameWithMeasuringPeriod(metric_name, measuring_period);
+  // If the metric plugin does not support WithTime implementation (i.e. is using the default
```

Comment:
Looks like this comment refers to the old name when it contained the words "WithTime".

Reply:
Good catch. Done.